### PR TITLE
boards: arm: efr32_radio: Add I2C0 node

### PR DESCRIPTION
**Pull request by ghost**
Add i2c0 pincntrl for efr32_radio_brd4161a

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]
Needs porting to hwmv2

File: boards/arm/efr32_radio/efr32_radio_brd4161a.dts
Comment (fkokosinski) [inline boards/arm/efr32_radio/efr32_radio_brd4161a.dts:73]: Using `location-*` properties is deprecated. You should convert these changes so that they use the Pinctrl API: https://docs.zephyrproject.org/latest/hardware/pinctrl/index.html

**Review by fkokosinski — CHANGES_REQUESTED** [review]

File: boards/arm/efr32_radio/efr32_radio_brd4161a.dts
Comment (Arunmanialagarsamy) [reply boards/arm/efr32_radio/efr32_radio_brd4161a.dts:73]: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/i2c/i2c_gecko.c
They didnt use the pinctrl api.
Do you want me to implement pinctrl apis in i2c_gecko.c driver?

Comment (fkokosinski) [reply boards/arm/efr32_radio/efr32_radio_brd4161a.dts]: That would be required for the Gecko I2C driver to use pinmuxing. You could base your work off of the Gecko UART driver: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_gecko.c

File: drivers/i2c/i2c_gecko.c
Comment (gmarull) [inline drivers/i2c/i2c_gecko.c:37]: please, make pinctrl mandatory, old methods should actually be removed as deprecation period is gone.

**Review by fkokosinski — CHANGES_REQUESTED** [review]
These changes seem to introduce a regression for me. Tested with:

```
west build -p -b efr32xg24_dk2601b samples/subsys/shell/shell_module -- -DCONFIG_I2C=y -DCONFIG_I2C_SHELL=y
```

With your changes, I get no UART output.

**Comment by fkokosinski** [comment]
I've unassigned myself since I'm no longer maintaining SiLabs platorms (see https://github.com/zephyrproject-rtos/zephyr/pull/71219).

File: drivers/i2c/i2c_gecko.c
Comment (gmarull) [inline drivers/i2c/i2c_gecko.c:157]: please, propagate error codes

**Review by gmarull — CHANGES_REQUESTED** [review]
split changes better, eg i2c driver changes into its own commit

**Comment by Arunmanialagarsamy** [comment]
@gmarull Gentle ping!

**Review by gmarull — APPROVED** [review]

**Comment by Arunmanialagarsamy** [comment]
@tgorochowik @fkokosinski waiting for your approval, please review it

**Review by fkokosinski — APPROVED** [review]

**Pull request merged**
